### PR TITLE
Issue #1841: Add icons to the admin bar.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1841

The one part of the PR that could use work is the addition of the classes. I'm currently adding a class to every link based on it's URL. It might be smarter to only add classes where they are needed (top-level only) and maybe name them something different? I liked these names because the paths will never change, where something like mlid could be different on every site.

### CODE REVIEW

File: core/modules/admin_bar/css/admin_bar.css [hunk 1]
```diff
@@ -137,7 +215,7 @@
 [dir="rtl"] #admin-bar .dropdown li li a {
   border-left: none;
 }
-#admin-bar .dropdown a:hover {
+#admin-bar .dropdown ul a:hover {
```
Comment (opi) [inline core/modules/admin_bar/css/admin_bar.css]: This prevent "user" and "logout" links to have a hover style.
Moreover, every other link (nested, nested expandable, ...) have a `text-decoration: underline;`, so I'd like first level link to have it too. #a11y

**Review by opi — CHANGES_REQUESTED** [review]

**Comment by docwilmot** [comment]
The icon for the Admin user looks funny on mobile. And I would like icons on mobile as well. Why not?

File: core/modules/admin_bar/css/admin_bar.css (hunk 1 above)
Comment (jenlampton) [reply core/modules/admin_bar/css/admin_bar.css]: I did that on purpose (removed the underline) as I thought it was a visual improvement. I didn't realize it was an #a11y regression -- putting back!

**Comment by jenlampton** [comment]
Added icons back for mobile.

> I'm currently adding a class to every link

Fixed this too.

**Comment by jenlampton** [comment]
Closing for new sandbox.

**Comment by jenlampton** [comment]
Sandbox?

**Comment by docwilmot** [comment]
`#admin-bar-menu.dropdown li > span.menu {    background: pink` ??

File: core/modules/admin_bar/admin_bar.theme.inc [hunk 2]
```diff
@@ -65,6 +65,12 @@ function theme_admin_bar_links($variables) {
       }
       if (isset($elements[$path]['#attributes']['class'])) {
         $elements[$path]['#attributes']['class'] = $elements[$path]['#attributes']['class'];
+
+        // Only add additional classes to top-level links.
+        if (substr_count($path, '/') <=1 ) {
```
Comment (klonos) [inline core/modules/admin_bar/admin_bar.theme.inc]:
```suggestion
        if (substr_count($path, '/') <= 1 ) {
```

File: core/modules/admin_bar/css/admin_bar.css [hunk 3]
```diff
@@ -68,14 +68,142 @@
...
+  background-size: 16px;
+}
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-content,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-people,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-appearance,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-modules,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-structure,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-config,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-devel,
+[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a.admin-reports {
```
Comment (klonos) [inline core/modules/admin_bar/css/admin_bar.css]: ...wondering whether we could replace all these with a single `[dir="rtl"] #admin-bar #admin-bar-menu > li > .dropdown > li > a`. This list here could get pretty huge and unmanageable if we decide to add icons to more menu items.

File: core/modules/admin_bar/admin_bar.theme.inc (hunk 2 above)
Comment (jenlampton) [reply core/modules/admin_bar/admin_bar.theme.inc]: Fixed!

File: core/modules/admin_bar/css/admin_bar.css (hunk 3 above)
Comment (jenlampton) [reply core/modules/admin_bar/css/admin_bar.css]: I didn't think that would work due to selector specificity, but it totally did! 
screen shot 2019-01-11 at 3 57 30 pm

File: core/modules/admin_bar/css/admin_bar.css [hunk 4]
```diff
@@ -68,14 +68,135 @@
...
+}
+[dir="rtl"] #admin-bar .dropdown .admin-bar-account > a {
+  background-position: 10% center;
+  padding-right: 10px;
+  padding-left: 28px;
+}
+
+/* Log Out link */
+#admin-bar .dropdown .admin-bar-logout > a {
+  background: transparent url(../images/sign-out--white--64.png) 95% center no-repeat; /* LTR */
```
Comment (opi) [inline core/modules/admin_bar/css/admin_bar.css]: background-position should be "90% center" to match other links in the #admin-bar-extra list.

**Comment by jenlampton** [comment]
This link is narrower. The icon position looked better at this position.

**Comment by opi** [comment]
> This link is narrower. The icon position looked better at this position.

Not in mobile view, though. Maybe 95% for all ?

**Comment by quicksketch** [comment]
Seems like part of the problem here is that the icons are different sizes, which is odd considering they're all named `--64.png`, which I would have thought meant 64x64 pixels. I think a better solution to using a background-image directly on the links would be to use a pseudo-element that is placed with `:after`, then it can be made a consistent size and not have to worry about percentage differences.

But in any case, I think that can happen later. For now I went ahead and went with the 90% across the board as suggested by opi, which looks better on mobile where they're all lined up vertically and doesn't make a discernible difference on desktop (to my eye). Let's go with this and refine as needed.

**Comment by opi** [comment]
> a better solution to using a background-image directly on the links would be to use a pseudo-element 

+1000 ; I keep this tasks in my head, for a rainy day (which come really often in Brittany, France).

thanks @quicksketch for the merge

**Pull request merged**